pkg/clients/aws.go

```diff
@@ -102,12 +105,35 @@ func UseProviderConfig(ctx context.Context, c client.Client, mg resource.Managed
 		if err := c.Get(ctx, types.NamespacedName{Namespace: csr.Namespace, Name: csr.Name}, s); err != nil {
 			return nil, errors.Wrap(err, "cannot get credentials secret")
 		}
-		return UseProviderSecret(ctx, s.Data[csr.Key], DefaultSection, region)
+		cfg, err := UseProviderSecret(ctx, s.Data[csr.Key], DefaultSection, region)
+		return SetResolver(ctx, mg, cfg), err
 	default:
 		return nil, errors.Errorf("credentials source %s is not currently supported", s)
 	}
 }
 
+// SetResolver parses annotations from the managed resource
+// and returns a configuration accordingly.
+func SetResolver(ctx context.Context, mg resource.Managed, cfg *aws.Config) *aws.Config {
+	if URL, ok := mg.GetAnnotations()["aws.crossplane.io/endpointURL"]; ok {
+		if SigningRegion, ok := mg.GetAnnotations()["aws.crossplane.io/endpointSigningRegion"]; ok {
```

Comment:
What would you think about making these annotations `aws.alpha.crossplane.io` since this is a new feature?

Reply:
Works for me. What do you think about using an annotation object rather than several annotations?

Reply:
I think it is _generally_ recommended to use separate annotations

Reply:
Ok then I'll leave it as is